.github/workflows/fe-build-and-test.yaml

```diff
@@ -0,0 +1,36 @@
+name: FE Test and Build
+
+on:
+  push:
+    branches:
+      - '!master'
```

Comment:
do we have tests running in another CI in master as of today (can't see it here: https://github.com/startreedata/thirdeye/blob/e35fbd77a91af0655751a2370ceb4b822252c292/.github/workflows/publish-frontend.yaml#L47)? 
if not I think we should also run in master - we will have tests for the edge cases of: 
- user forcing a PR to be merged before PR tests pass
- user pushing to master directly
the CI should be cheap so no need to over optimize here, I'd rather be on the safe side.